src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -13,25 +14,19 @@ namespace Aspire.Hosting.Azure;
 
 /// <summary>
 /// Represents the infrastructure for Azure Container Apps within the Aspire Hosting environment.
-/// Implements the <see cref="IDistributedApplicationLifecycleHook"/> interface to provide lifecycle hooks for distributed applications.
 /// </summary>
 internal sealed class AzureContainerAppsInfrastructure(
     ILogger<AzureContainerAppsInfrastructure> logger,
-    IOptions<AzureProvisioningOptions> provisioningOptions,
-    DistributedApplicationExecutionContext executionContext) : IDistributedApplicationLifecycleHook
+    DistributedApplicationExecutionContext executionContext,
+    IOptions<AzureProvisioningOptions> options) : IDistributedApplicationEventingSubscriber
 {
-    public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    public async Task OnBeforeStartAsync(BeforeStartEvent @event, CancellationToken cancellationToken = default)
```

Comment:
Does this method still need to be `public`? I think it is only referenced by this class now.

Reply:
I've updated this (and a few other instances that aren't used in tests) to private.